src/cascadia/TerminalSettingsEditor/Profiles.cpp

```diff
@@ -221,13 +416,54 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         const auto backgroundImgCheckboxTooltip{ ToolTipService::GetToolTip(UseDesktopImageCheckBox()) };
         Automation::AutomationProperties::SetFullDescription(UseDesktopImageCheckBox(), unbox_value<hstring>(backgroundImgCheckboxTooltip));
 
+        const auto showAllFontsCheckboxTooltip{ ToolTipService::GetToolTip(ShowAllFontsCheckbox()) };
+        Automation::AutomationProperties::SetFullDescription(ShowAllFontsCheckbox(), unbox_value<hstring>(showAllFontsCheckboxTooltip));
+
         Automation::AutomationProperties::SetName(DeleteButton(), RS_(L"Profile_DeleteButton/Text"));
     }
 
+    IInspectable Profiles::CurrentFontFace() const
+    {
+        // look for the current font in our shown list of fonts
+        const auto& profileVM{ State().Profile() };
+        const auto profileFontFace{ profileVM.FontFace() };
+        const auto& currentFontList{ profileVM.ShowAllFonts() ? profileVM.CompleteFontList() : profileVM.MonospaceFontList() };
+        IInspectable fallbackFont;
+        for (const auto& font : currentFontList)
+        {
+            if (font.LocalizedName() == profileFontFace)
+            {
+                return box_value(font);
+            }
+            else if (font.LocalizedName() == L"Cascadia Mono")
+            {
+                fallbackFont = font;
+            }
+        }
+
+        // we couldn't find the desired font, set to "Cascadia Mono" since that ships by default
+        return fallbackFont;
+    }
+
+    void Profiles::FontFace_SelectionChanged(IInspectable const& /*sender*/, SelectionChangedEventArgs const& e)
+    {
+        // NOTE: We need to hook up a selection changed event handler here instead of directly binding to the profile view model.
+        //       A two way binding to the view model causes an infinite loop because both combo boxes keep fighting over which one's right.
+        const auto selectedItem{ e.AddedItems().GetAt(0) };
+        const auto newFontFace{ unbox_value<Editor::Font>(selectedItem) };
+        State().Profile().FontFace(newFontFace.LocalizedName());
+    }
+
     void Profiles::OnNavigatedTo(const NavigationEventArgs& e)
     {
         _State = e.Parameter().as<Editor::ProfilePageNavigationState>();
 
+        // generate the font list, if we don't have one
+        if (!_State.Profile().CompleteFontList() || !_State.Profile().MonospaceFontList())
+        {
+            ProfileViewModel::UpdateFontList();
```

Comment:
actually wait this _is_ done in the ctor. So is this ever hit?

Reply:
I only have this here in case loading the fonts fails for some reason. The one in the constructor is hit when the Settings UI is opened, but then we check if we have it when we navigate to the profiles page.